tools/c7n_mailer/c7n_mailer/ldap_lookup.py

```diff
@@ -26,7 +27,17 @@
 
 class LdapLookup(object):
 
-    def __init__(self, config, logger):
+    def __init__(self, config, session, logger):
```

Comment:
rather than pass the aws client session into the ldap, it would be better separation of concerns to have the caller decrypt and just pass the ldap config to this.